### PR TITLE
Add v25.0.0-alpha.0 release for AWS (CAPA)

### PR DESCRIPTION
**Pull request by nprokopic**
Towards https://github.com/giantswarm/roadmap/issues/3466

### What is new?

- `v25.0.0-alpha.0` release for AWS (CAPA) which should have app and component versions same as the (current) cluster-aws release `v0.76.2`.
- CI config to push new AWS (CAPA) releases to CAPA app collection (which effectively delivers published/merged releases to MCs).
- CI config to create and update GitHub releases for added and WIP (open PRs) AWS (CAPA) releases.
  - While getting Releases from the MC is usually enough, having GitHub releases enables easier consumption of releases (especially the WIP releases for open PRs) in our CI and other components (e.g. kubectl-gs).
  - Releases `Client` and `Builder` from Releases SDK are getting Release CRs from GitHub releases.
  - e.g. For the newly added AWS (CAPA) release from this PR we now have this draft GitHub release https://github.com/giantswarm/releases/releases/tag/untagged-a92968cbb4ae0b98d1e2.

### Checklist
- [x] Roadmap issue created
- [ ] Release uses same Flatcar version like the cluster-aws release v0.76.2.
- [ ] Release uses same Kubernetes patch version like the cluster-aws release v0.76.2.

### CODE REVIEW

**Review by Gacko — COMMENTED** [review]
I can't tell a lot about the actual app & component versions being used, but is `v25.0.0-alpha.1` semver compliant? I'm not used to a fourth level in version semantics, rather having alpha and beta versions written in the form of `<major>.<minor>.<patch>-alpha<number>`, so for example as follows:

- v25.0.0-alpha1
- v25.0.0-alpha2
- v25.0.0-alpha3
- v25.0.0-beta1
- v25.0.0-beta2
- v25.0.0-beta3
- v25.0.0-rc1
- v25.0.0-rc2
- v25.0.0-rc3
- v25.1.0-alpha1
- v25.1.0-alpha2
- v25.1.0-alpha3
- v25.1.0-beta1
- v25.1.0-beta2
- v25.1.0-beta3
- v25.1.0-rc1
- v25.1.0-rc2
- v25.1.0-rc3
- v26.0.0-alpha1
- v26.0.0-alpha2
- v26.0.0-alpha3
- v26.0.0-beta1
- v26.0.0-beta2
- v26.0.0-beta3
- v26.0.0-rc1
- v26.0.0-rc2
- v26.0.0-rc3
- ... and so on.

File: capa/v26.0.0-alpha.1/release.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  - name: security-bundle
+    version: 1.6.5
+    catalog: giantswarm
+    dependsOn:
+    - prometheus-operator-crd
+  - name: teleport-kube-agent
+    version: 0.9.0
+  components:
+  - name: cluster-aws
+    version: 0.76.1-b76af2c26f4224ffb0d718e940e232fac05c89a0
```
Comment (Gacko) [inline capa/v26.0.0-alpha.1/release.yaml:93]: I know, this is mostly a demo, but would this work? Can we reference apps from test catalogs?

File: capa/v26.0.0-alpha.1/release.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    - prometheus-operator-crd
+  - name: cilium
+    version: 0.24.0
+  - name: cilium-crossplane-resources
+    version: 0.1.0
+  - name: cilium-servicemonitors
+    version: 0.1.2
+    dependsOn:
+    - prometheus-operator-crd
+  - name: cloud-provider-aws
```
Comment (Gacko) [inline capa/v26.0.0-alpha.1/release.yaml:37]: Does this refer to https://github.com/giantswarm/cluster-aws/blob/1c557003b5f5b3c1316ba17b6d5f88c24d55cbcd/helm/cluster-aws/templates/cloud-provider-aws-helmrelease.yaml#L19?

File: capa/v26.0.0-alpha.1/release.yaml (hunk 1 above)
Comment (T-Kukawka) [reply capa/v26.0.0-alpha.1/release.yaml:93]: yeah that works on Vintage as well

**Comment by nprokopic** [comment]
> is v25.0.0-alpha.1 semver compliant?

Yes, everything after the dash `-` is a single string.

e.g. Kubernetes is doing it like this

image

**Comment by nprokopic** [comment]
Also CAPI 
image

**Comment by Gacko** [comment]
Okay, thanks for clarifying that. I didn't know Kubernetes does that, only knew some other projects where they are using it as I described above.

File: capa/v26.0.0-alpha.1/release.yaml (hunk 2 above)
Comment (nprokopic) [reply capa/v26.0.0-alpha.1/release.yaml]: yes

**Comment by nprokopic** [comment]
And draft release is gone :tada:

image

Reopening this now.

**Review by T-Kukawka — APPROVED** [review]

**Review by weseven — APPROVED** [review]

**Pull request merged**